frontstage_api/Resources/secure_messaging.py

```diff
@@ -17,10 +17,105 @@ class GetMessagesList(Resource):
 
     @staticmethod
     def get(encoded_jwt):
-        messages = secure_messaging_controllers.get_messages_list(encoded_jwt)
-        unread_messages_total = secure_messaging_controllers.get_unread_message_total(encoded_jwt)
-        messages_list = {**messages, **unread_messages_total}
-        return make_response(jsonify(messages_list), 200)
+        label = request.args.get('label')
+
+        messages = secure_messaging_controllers.get_messages_list(encoded_jwt, label)
+        # If the messages were returned with errors return the error message
+        if messages.get('error'):
+            return make_response(jsonify(messages), 200)
+
+        messages["unread_messages_total"] = secure_messaging_controllers.get_unread_message_total(encoded_jwt)
+
+        return make_response(jsonify(messages), 200)
+
+
+class GetMessageView(Resource):
```

Comment:
Would it improve the readability if we split the resources out into separate files?

Reply:
I quite like this idea, would keep things a lot tidier